tools/corals/multilocus_genotype/multilocus_genotype.R

```diff
@@ -57,42 +64,48 @@ vcf <- read.vcfR(opt$input_vcf);
 
 # Convert VCF file into a genind for the Poppr package.
 # TODO: probably should not hard-code 2 cores.
-gl <- vcfR2genlight(vcf, n.cores=2);
-gind <- new("genind", (as.matrix(gl)));
+# changed to genind format for extracting alleles later
+# trade-off is it is a bit slower to import data
+# gl <- vcfR2genlight(vcf, n.cores=2)
+# gind <- new("genind", (as.matrix(gl)))
+gind <- vcfR2genind(vcf);
 
 # Add population information to the genind object.
 poptab <- read.table(opt$input_pop_info, check.names=FALSE, header=F, na.strings=c("", "NA"), stringsAsFactors=FALSE, sep="\t");
 colnames(poptab) <- c("row_id", "affy_id", "user_specimen_id", "region");
 gind@pop <- as.factor(poptab$region);
+strata(gind)<-data.frame(pop(gind));
 
 # Convert genind object to a genclone object.
 obj2 <- as.genclone(gind);
 
 # Calculate the bitwise distance between individuals.
 xdis <- bitwise.dist(obj2);
 
-# Multilocus genotypes (threshold of 16%).
-mlg.filter(obj2, distance=xdis) <- 0.016;
+# Multilocus genotypes (threshold of 3.2%).
+# threshold doubled because of how the data is formatted in genind compared to genlight
+mlg.filter(obj2, distance=xdis) <- 0.032;
 m <- mlg.table(obj2, background=TRUE, color=TRUE);
 
 # Create table of MLGs.
 id <- mlg.id(obj2);
-dt <- data.table(id, keep.rownames=TRUE);
-setnames(dt, c("id"), c("affy_id"));
+#dt <- data.table(id, keep.rownames=TRUE);
```

Comment:
These commented code lines (92, 93) should be removed instead of commenting them out to keep the code clean.